### ISSUE TITLE
$collation argument: Unification

### ISSUE THREAD

**Issue by ChristianGruen** [qt4cg/qtspecs#343]
In the function set of the XQFO current specification, the type of the `$collation` parameter is sometimes `xs:string` and sometimes `xs:string?`, depending on the position of the parameter. Examples:

#### Mandatory

```xquery
fn:distinct-values($values as xs:anyAtomicType*, $collation as fn:default-collation()) as xs:anyAtomicType*
fn:index-of($input | as xs:anyAtomicType*, $search | as xs:anyAtomicType, $collation as xs:string) as xs:integer*
```

#### Optional

```xquery
fn:sort($input as item()*, $collation as xs:string?, $key as function(item()) as xs:anyAtomicType*
fn:lowest($input as item()*, $collation as xs:string?, $key as function(item()) as xs:anyAtomicType*) as item()*
```

I think we should always allow an empty sequence.

**Comment by rhdunn** [comment]
From the closed #70 issue:
1. Change `fn:collation-key`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:compare`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:contains`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:contains-token`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:deep-equal`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:differences`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:distinct-values`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:ends-with`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:index-of`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:max`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:min`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:starts-with`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:substring-after`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:substring-before`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:uniform`/`$collation` from `xs:string` to `xs:string?`.
1. Change `fn:unique`/`$collation` from `xs:string` to `xs:string?`.

**Comment by ChristianGruen** [comment]
Thanks; I missed that one.
